src/ChatFeed/MessageList/Message/index.tsx

```diff
@@ -0,0 +1,254 @@
+import React, { useState } from 'react';
+
+import { Props } from './props';
+import { myStyles, theirStyles } from './styles';
+
+import { Dot } from '../../../Dot';
+import { Avatar } from '../../../Avatar';
+
+import { DateTime } from './DateTime';
+
+import { Attachment } from './Attachment';
+
+import { isImage, getFileName } from '../../../util/file';
+import { formatTime, getDateTime } from '../../../util/dateTime';
+
+import { Row, Col, setConfiguration } from 'react-grid-system';
+
+setConfiguration({ maxScreenClass: 'xl' });
+
+export const Message: React.FC<Props> = ({
+  lastMessage = null,
+  message,
+  nextMessage = null,
+  chat = null,
+  isSending = false,
+  isMyMessage = false,
+  showDateTime = false,
+  style = {},
+}) => {
+  const [hovered, setHovered] = useState<boolean>(false);
+
+  const topRadius =
+    !lastMessage || lastMessage.sender_username !== message.sender_username
+      ? '1.3em'
+      : '0.3em';
+  const bottomRadius =
+    !nextMessage || nextMessage.sender_username !== message.sender_username
+      ? '1.3em'
+      : '0.3em';
+  const borderRadius = isMyMessage
+    ? `1.3em ${topRadius} ${bottomRadius} 1.3em`
+    : `${topRadius} 1.3em 1.3em ${bottomRadius}`;
+  const paddingBottom =
+    !nextMessage || nextMessage.sender_username !== message.sender_username
+      ? '12px'
+      : '2px';
+  const styles = isMyMessage ? myStyles : theirStyles;
+  const sendingStyle = isSending
+    ? {
+        backgroundColor: isSending ? '#40a9ff' : 'rgb(24, 144, 255)',
+      }
+    : {};
+
+  const text: string =
+    message.text !== null
+      ? message.text.replace(/<a /g, `<a style="color: 'white';" `)
+      : '';
+
+  const renderImages = () => {
+    const attachments =
+      message && message.attachments ? message.attachments : [];
+
+    return attachments.map((attachment, index) => {
+      const fileName = getFileName(attachment.file);
+
+      if (isImage(fileName)) {
+        return (
+          <Attachment
+            attachment={attachment}
+            isLoading={isSending || attachment.file === null}
+            style={{ ...styles.image, ...style.image }}
+          />
+        );
+      }
+
+      return <div key={`attachment${index}`} />;
+    });
+  };
+
+  const renderFiles = () => {
+    const attachments =
+      message && message.attachments ? message.attachments : [];
+
+    return attachments.map((attachment, index) => {
+      const fileName = getFileName(attachment.file);
+
+      if (!isImage(fileName)) {
+        return (
+          <Attachment
+            attachment={attachment}
+            isLoading={isSending || attachment.file === null}
+            style={{ ...styles.file, ...style.file }}
+          />
+        );
+      }
+
+      return <div key={`attachment${index}`} />;
+    });
+  };
```

Comment:
maybe combine these methods and pass an boolean to pick if it should render images and files- other than the if clause and style they are identical

Reply:
Happy to do that - love chopping line count down 👍